tests/integration-tests/tests/common/schedulers_common.py

```diff
@@ -382,13 +382,15 @@ def assert_job_state(self, job_id, expected_state):  # noqa: D102
     def compute_nodes_count(self, filter_by_partition=None):  # noqa: D102
         return len(self.get_compute_nodes(filter_by_partition))
 
-    def get_compute_nodes(self, filter_by_partition=None):  # noqa: D102
-        command = "sinfo --Node --noheader --responding"
+    def get_compute_nodes(self, filter_by_partition=None, all_nodes=None):  # noqa: D102
+        command = "sinfo --Node --noheader"
         if filter_by_partition:
             command += " --partition {}".format(filter_by_partition)
         # Print first and fourth columns to get nodename and state only (default partition contains *)
         # Filter out nodes that are not responding or in power saving states
-        command += " | awk '{print $1, $4}' | grep -v '[*#~%]' | awk '{print $1}'"
+        command += (
+            " | awk '{print $1, $4}' | grep -v '[*#~%]' | awk '{print $1}'" if not all_nodes else " | awk '{print $1}'"
+        )
```

Comment:
I'm not sure leaving the `| grep -v '[*#~%]'` is equivalent to having `--responding` in the `command` string.

In my [previous comment](https://github.com/aws/aws-parallelcluster/pull/4258#discussion_r942723551) I meant *moving* the `--responding` option into the if condition like this:
```python
        command += (
            " --responding | awk '{print $1, $4}' | grep -v '[*#~%]' | awk '{print $1}'" if not all_nodes else " | awk '{print $1}'"
        )
```

Reply:
We just checked the sinfo state codes, and we saw that `*` means non-responding. So the `grep -v '[*#~%]'` is already filtering out non-responding nodes. So we can safely remove `--responding` from the initial `command` string.